### DISCUSSION TITLE
Add gif support to supabase image transformations - or specify in docs that it is not supported.

### DISCUSSION THREAD

**Discussion by jonathanlal** [supabase/supabase#25951] [Feature Requests]
Currently if you try and transform a .gif file it will make the gif static (by using only the first frame). Which is the default behaviour of imgproxy. 

```
  const {
    data,
  } = supabaseClient.storage.from('bucket').getPublicUrl(
    'someGifFile',
    {
      transform: {
        width: 100,
        height: 100,
      },
    },
  );
```

imgproxy does support transforming more than 1 frame in a gif file but requires changing an environment variable:

[IMGPROXY_MAX_ANIMATION_FRAMES](https://docs.imgproxy.net/configuration/options#IMGPROXY_MAX_ANIMATION_FRAMES)

Perhaps passing `format: 'gif'` in the transform options could allow for it?

If not it should be at least updated in the docs that gif files are not supported in image transformations. Since it shows misleading information:
https://supabase.com/docs/guides/storage/serving/image-transformations

![image](https://github.com/supabase/supabase/assets/13243000/4c188351-ec44-4e9b-b3dc-5adcafdb03d1)

![image](https://github.com/supabase/supabase/assets/13243000/1f04d0c0-3439-480e-9c5a-ca20795cf027)

**Comment by y4my4my4m** [comment]
Please implement this supabase staff, it's already supported by IMGPROXY is just requires passing the env to imgproxy

**Reply by y4my4my4m** [reply]
@jonathanlal 
actually, i got it to work on the self-hosted

you can add this to your docker-compose.yml
      IMGPROXY_MAX_ANIMATION_FRAMES: 120

it can't be set to `0` though.

**Comment by dpjnguyen** [comment]
We need IMGPROXY_MAX_ANIMATION_FRAMES with hosted supabase to be at least 150 frames or configurable.
